packages/p2p-media-loader-shaka/src/services/segment-manager.ts

```diff
@@ -1,57 +1,40 @@
-import { Segment, Stream } from "./segment";
-import { HookedStream, StreamInfo, StreamType } from "../types/types";
+import * as Utils from "./stream-utils";
+import { HookedStream, StreamInfo, Stream } from "../types/types";
+import {
+  Core,
+  ReadonlyStream,
+  Segment,
+  StreamType,
+} from "p2p-media-loader-core";
 
 export class SegmentManager {
-  private manifestUrl?: string;
-  readonly streams: Map<number, Stream> = new Map();
-  readonly urlStreamMap: Map<string, Stream> = new Map();
-  readonly streamInfo: StreamInfo;
+  private readonly streamInfo: Readonly<StreamInfo>;
+  private readonly core: Core<Stream>;
 
-  constructor(streamInfo: StreamInfo) {
+  constructor(streamInfo: Readonly<StreamInfo>, core: Core<Stream>) {
     this.streamInfo = streamInfo;
+    this.core = core;
   }
 
-  setManifestUrl(url: string) {
-    this.manifestUrl = url.split("?")[0];
-  }
-
-  setStream({
-    stream,
-    streamOrder = -1,
-    bitrate,
-  }: {
-    stream: HookedStream;
-    streamOrder?: number;
-    bitrate: number;
-  }) {
-    if (!this.manifestUrl || this.streams.has(stream.id)) return;
-
-    const managerStream = new Stream({
-      localId: stream.id,
-      order: streamOrder,
-      type: stream.type as StreamType,
-      manifestUrl: this.manifestUrl,
+  setStream(stream: HookedStream, type: StreamType, index = -1) {
+    const managerStream: Stream = {
+      localId: stream.id.toString(),
+      type,
       url: stream.streamUrl,
```

Comment:
Can we use `stream.streamUrl` as `localId`? This way, `url` can be removed from `Stream` type and `getStreamByUrl` removed from the `Core`.